components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/add-security-questions.jsp

```diff
@@ -0,0 +1,145 @@
+  ~ under the License.
+  --%>
+<%@ page contentType="text/html;charset=UTF-8" pageEncoding="UTF-8" %>
+
+<%@ page import="org.owasp.encoder.Encode" %>
+
+<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.AuthenticationEndpointUtil" %>
+<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.EncodedControl" %>
+<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.Constants" %>
+<%@ page import="org.wso2.carbon.identity.recovery.ui.IdentityManagementAdminClient" %>
+<%@ page import="org.wso2.carbon.identity.recovery.model.ChallengeQuestion" %>
+<%@ page import="org.wso2.carbon.utils.ServerConstants" %>
+<%@ page import="org.wso2.carbon.ui.CarbonUIUtil" %>
+<%@ page import="org.wso2.carbon.CarbonConstants" %>
+
+<%@ page import="java.nio.charset.StandardCharsets" %>
+<%@ page import="java.util.ResourceBundle" %>
+<%@ page import="java.util.ArrayList" %>
+<%@ page import="java.util.Iterator" %>
+<%@ page import="java.util.HashMap" %>
+<%@ page import="java.util.List" %>
+<%@ page import="java.util.Map" %>
+
+<%
+    String BUNDLE = "org.wso2.carbon.identity.application.authentication.endpoint.i18n.Resources";
+    ResourceBundle resourceBundle = ResourceBundle.getBundle(BUNDLE, request.getLocale(), new
+            EncodedControl(StandardCharsets.UTF_8.toString()));
+    String urlData = request.getParameter("data");
+    // Extract the challenge questions from the request and add them into an array
+    String[] questionSets = null;
+    if (urlData != null) {
+        questionSets = urlData.split("&");
+    }
+    // Hash-map to hold available challenge questions in the system
+    Map<String, List<ChallengeQuestion>> challengeQuestionMap = new HashMap<>();
+
+    for (String question : questionSets) {
+        String[] questionProperties = question.split("\\|");
+        // Construct a new ChallengeQuestion for each challenge question received from the request
+        ChallengeQuestion tempChallengeQuestion = new ChallengeQuestion();
+        // Extract the challenge question properties
+        String questionSetId = questionProperties[0];
+        String questionId = questionProperties[1];
+        String questionBody = questionProperties[2];
+        tempChallengeQuestion.setQuestionSetId(questionSetId);
+        tempChallengeQuestion.setQuestionId(questionId);
+        tempChallengeQuestion.setQuestion(questionBody);
+        // Add the challenge question to the Hash-map
+        List<ChallengeQuestion> challengeQuestionList = challengeQuestionMap.get(questionProperties[0]);
+        if (challengeQuestionList == null) {
+            challengeQuestionList = new ArrayList<>();
+            challengeQuestionList.add(tempChallengeQuestion);
+            challengeQuestionMap.put(questionProperties[0], challengeQuestionList);
+        } else {
+            challengeQuestionList.add(tempChallengeQuestion);
```

Comment:
What is the need of this else part?. Even value was added to challengeQuestionList, it is not used.

Reply:
We need that part,  challengeQuestionList is a reference and we are adding to that reference. That will be used by a latter part of the code